src/PubSub/Connection/Grpc.php

```diff
@@ -288,9 +290,12 @@ public function acknowledge(array $args)
      */
     public function listSnapshots(array $args)
     {
-        return $this->send([$this->subscriberClient, 'listSnapshots'], [
-            $this->pluck('project', $args),
-            $args
+        return $this->whitelist([$this, 'send'], [
```

Comment:
This may not fit with the approach taken elsewhere, but what do you think about having the `whitelist` method return a callable? That could potentially mitigate needing to pass multiple nested `[class, method]` arrays as args to a single function. So this code would look like:

```
$sendCallable = $this->whitelist([$this, 'send']);
return $callable([$this->subscriberClient, 'listSnapshots'], [
  $this->pluck('project', $args),
  $args
]);
```

I don't necessarily think this is better (it makes more sense if `$this->send` also returns a callable) but thought I would mention it.